src/neo-vm/Helper.cs

```diff
@@ -5,25 +5,11 @@ namespace Neo.VM
 {
     internal static class Helper
     {
-        public static byte[] ReadVarBytes(this BinaryReader reader, int max = 0X7fffffc7)
+        public static byte[] SafeReadBytes(this BinaryReader reader, int max = 0x1000000)
         {
-            return reader.ReadBytes((int)reader.ReadVarInt((ulong)max));
-        }
-
-        public static ulong ReadVarInt(this BinaryReader reader, ulong max = ulong.MaxValue)
-        {
-            byte fb = reader.ReadByte();
-            ulong value;
-            if (fb == 0xFD)
-                value = reader.ReadUInt16();
-            else if (fb == 0xFE)
-                value = reader.ReadUInt32();
-            else if (fb == 0xFF)
-                value = reader.ReadUInt64();
-            else
-                value = fb;
-            if (value > max) throw new FormatException();
-            return value;
+            if((max > 0x1000000) || (!reader.BaseStream.CanSeek) || (reader.BaseStream.Length - reader.BaseStream.Position) < max))
```

Comment:
If we want to throw an exception, then is not needed to check `CanSeek`, because this could throw an exception when you access to length in wrong streams